### ISSUE TITLE
Use Prebuilt contrib's for CI

### ISSUE THREAD

**Issue by poshul** [OpenMS/OpenMS#6451]
We currently pull down the sources for the contribs and build them each time we run Jenkins. Since we already have a job to independently build the binaries (which are stored https://abibuilder.cs.uni-tuebingen.de/archive/openms/contrib/) it would be more efficient to use those rather than build each time CI is run.

This would consist of updating our https://abibuilder.cs.uni-tuebingen.de/jenkins/job/openms/job/contrib/job/openms_contrib_push/ to store each build based on its hash (with some logic to delete old versions), and updating our other CI jobs to check if there is a binary package for the contrib it's OpenMS branch requires, and if so pull that instead of building.

**Comment by timosachsenberg** [comment]
Would vcpkg be an option here? A friend uses it with quite some success.

**Comment by jpfeuffer** [comment]
Yes. Or conan + artifactory. But they come with their own complexity and problems. See the issues from my profile against those tools.

**Comment by timosachsenberg** [comment]
just throwing that in. this friend liked how easy vcpkg worked for him (he just caches the artefacts on his local server)

**Comment by timosachsenberg** [comment]
see https://github.com/OpenMS/OpenMS/pull/7236
